src/utils/errorHandling.ts

```diff
@@ -0,0 +1,32 @@
+import type { Success, Failure, Result } from '@/shared.types';
+
+function createSuccessResult<T>(data?: T): Success<T> {
```

Comment:
Since these functions will be used frequently, I would recommend choosing less verbose names to improve readability, such as `success()` and `failure()`.